### ISSUE TITLE
Keras callback support

### ISSUE THREAD

**Issue by marcoancona** [IBM/adversarial-robustness-toolbox#221]
To my understanding, if we pass Keras callbacks during adversarial training, these are called at every epoch of the underlying model, ie. every training *batch*. This make it difficult to use the same callbacks, ie. `ModelCheckpoint` and `Tensorboard` for normal and adversarial traning. Some callbacks like `EarlyStopping` do not work at all, because there is nothing to stop within a single training epoch.

**Describe the solution you'd like**
Callbacks passed to `fit` and `fit_generator` of `AdversarialTrainer` should behave like callbacks passed to a Keras model directly. A possible solution is for `AdversarialTrainer` to trigger the callback (`on_epoch_begin`, `on_epoch_end`, etc) when appropriate.

**Comment by beat-buesser** [comment]
Hi @marcoancona Thank you very much for this interesting feature request. I think this could be very helpful for many users of ART. Have you already managed to implement the described solution for your experiment?

**Comment by marcoancona** [comment]
I did not have the time to try this myself. I only inspected the `AdversarialTrainer` to better understand why the callbacks were not behaving the way I expected. I believe this is very important. For example, at the moment I am running an adversarial training for 12 hours but I have no way to know:
- what is the current progress (how many epochs have been trained)
- what is the current train/test accuracy (I only see the training accuracy for each batch, which is rather useless)
![image](https://user-images.githubusercontent.com/2278117/69076793-8427cf00-0a34-11ea-9295-8c947482ac91.png)

**Comment by SilverKnightVGM** [comment]
Has there been any progress on this @marcoancona? I am also interested in using callbacks with the AdversarialTrainer.
